### DISCUSSION TITLE
treat polymorphic function with no value parameters as by-name

### DISCUSSION THREAD

**Discussion by bishabosha** [scala/scala3#17858] [Feature Requests]
## Reducing Boilerplate
```scala
val none: [T] => Option[T] =
  [T] => None
```
expands to
```scala
val none: [T] => () => Option[T] =
  [T] => () => None
```

with this inference, call sites would still look like the following:
```scala
val m = none[Int]() // : Option[Int]
val n = none() // : Option[Nothing]
val o = none[Int] // error - method apply must be called with () argument
```

**Comment by smarter** [comment]
Having parens at the use-site but not the definition site looks confusing and doesn't match any other construct in the language, if we do accept `[T] => Option[T]` I suggest treating it as a "polymorphic by-name", so `val n = none` by itself would have type `Option[Nothing]`, this is also similar to how context functions work.
